File: laravel/app/Http/Controllers/UserController.php
<?php

namespace App\Http\Controllers;

use App\User;
use App\Services\User as UserService;

class UserController extends Controller
{
  public function index(UserService $service)
  {
    $users = User::all();

    return view('users', compact('users'));

    //return $service->getAllUsers($this);
  }

  public function show($userId, UserService $service)
  {
    $user = User::find($userId);

    return view('user', compact('user'));

    //return $service->showUser($this, $userId);
  }

  public function showAllUsers(array $data)
  {
    return view('users', $data);
  }

  public function showUser(array $data)
  {
    return view('user', $data);
  }
}

File: devcon_elixir/lib/devcon_elixir/web/controllers/user_controller.ex
defmodule DevconElixir.Web.UserController do
  use DevconElixir.Web, :controller

  alias DevconElixir.{Repo, User}

  def index(conn, _params) do
    users = Repo.all(User)

    render(conn, "index.html", %{users: users})
  end

  def show(conn, %{"id" => user_id}) do
    user = Repo.get_by(User, id: user_id)

    render(conn, "show.html", %{user: user})
  end
end